x509/verify.go

```diff
@@ -173,6 +173,17 @@ func (c *Certificate) inChain(chain []*Certificate) bool {
 
 func (c *Certificate) buildChains(cache map[int][][]*Certificate, currentChain []*Certificate, opts *VerifyOptions) (chains [][]*Certificate, err error) {
 
+	// If the certificate being validated is a root, add the chain of length one
+	// containing just the root. Only do this on the first call to buildChains,
+	// when the len(currentChain) = 1.
```

Comment:
Which part of the _why_ is confusing? Is it why do return a chain of length 1, or why isn't this somehow handled later in the function?